Solutions/CohesitySecurity/Data Connectors/Helios2Sentinel/Cohesity_API_FunctionApp.json

```diff
@@ -0,0 +1,102 @@
+        },
+        {
+            "title": "",
+            "description": ">**(Optional Step)** Securely store workspace and API authorization key(s) or token(s) in Azure Key Vault. Azure Key Vault provides a secure mechanism to store and retrieve key values. [Follow these instructions](https://docs.microsoft.com/azure/app-service/app-service-key-vault-references) to use Azure Key Vault with an Azure Function App."
+        },
+        {
+            "title": "",
+            "description": "**STEP 1 - Get a Cohesity DataHawk API key (see troubleshooting [instruction 1](https://github.com/cohesity/Azure-Sentinel/tree/CohesitySecurity.internal/Solutions/CohesitySecurity/Data%20Connectors/Helios2Sentinel/IncidentProducer))**"
+        },
+        {
+            "title": "",
+            "description": "**STEP 2 - Register Azure app ([link](https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps)) and save Application (client) ID, Directory (tenant) ID, and Secret Value ([instructions](https://learn.microsoft.com/en-us/azure/healthcare-apis/register-application)). Grant it Azure Storage (user_impersonation), Microsoft Graph (Application.ReadWrite.All, User.Read) permissions**"
+        },
+        {
+            "title": "",
+            "description": "**STEP 3 - Choose ONE from the following two deployment options to deploy the connector and the associated Azure Functions**."
+        },
+        {
+            "title": "Option 1 - Azure Resource Manager (ARM) Template",
+            "description": "Use this method for automated deployment of the Cohesity data connector using an ARM Template.\n\n1. Click the **Deploy to Azure** button below. \n\n\t[![Deploy To Azure](https://aka.ms/deploytoazurebutton)](https://aka.ms/sentinel-Cohesity-azuredeploy)\n2. Select the preferred **Subscription**, **Resource Group** and **Location**. \n3. Enter the parameters that you created at the previous steps\n4. Mark the checkbox labeled **I agree to the terms and conditions stated above**.\n5. Click **Purchase** to deploy."
+        },
+        {
+            "title": "Option 2 - Manual Deployment of Azure Functions",
+            "description": "Use the following step-by-step instructions to deploy the Cohesity data connector manually with Azure Functions.\n\n1. Clone the github [repository](https://github.com/cohesity/Azure-Sentinel).\n2. Install the pre-requisites mentioned [here](https://github.com/cohesity/Azure-Sentinel/tree/CohesitySecurity.internal/Solutions/CohesitySecurity/Data%20Connectors/Helios2Sentinel).\n3. Edit [cohesity.json](https://github.com/cohesity/Azure-Sentinel/blob/CohesitySecurity.internal/Solutions/CohesitySecurity/cohesity.json) to add the required values.\n4. Run this [script](https://github.com/cohesity/Azure-Sentinel/blob/CohesitySecurity.internal/Solutions/CohesitySecurity/Data%20Connectors/Helios2Sentinel/deploy_config.sh)."
```

Comment:
@yinghuang123: In the manual deployment steps you are asking the author to clone entire Sentinel Repo, which we feel is not correct. Instead, you can refer instructions from other function app-based connectors to do the manual deployment and update them as needed. Also, please make sure all these urls are updated and pointing to Azure-Sentinel repo master branch.

**For Your Reference:** https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Agari/Data%20Connectors/Agari_API_FunctionApp.json

Reply:
We have removed the manual option as it's getting too complicated to list down all the required steps as we require creation of storage accounts, key vaults, providing many permissions before even starting to compilation. One other issue is that we require keyvault with a specific name to be present which is dependent on ARM template [function](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-string#uniquestring) which maybe difficult for the user to generate. For these reasons, we have removed the manual option.